components/Reference/DynamicFee.tsx

```diff
@@ -27,12 +27,20 @@ const DynamicFee = ({ referenceItem, fork }: Props) => {
 
   const { common } = useContext(EthereumContext)
   const [inputs, setInputs] = useState<InputValue | undefined>()
-  const [result, setResult] = useState('0')
+  const [gas_cost, setGasCost] = useState('0')
+  const [gas_refund, setGasRefund] = useState('0')
+  const [can_refund, setCanRefund] = useState(false)
 
   const handleCompute = debounce((inputs) => {
     if (common) {
       try {
-        setResult(calculateDynamicFee(referenceItem, common, inputs))
+        setGasCost(calculateDynamicFee(referenceItem, common, inputs))
+
+        const refund = calculateDynamicRefund(referenceItem, common, inputs)
+        if (refund != null) {
+          setCanRefund(true)
+          setGasRefund(refund)
+        }
```

Comment:
should there be a fallback to reset back to 0/false?

Reply:
It seems to have the wanted behaviour already, what would it change?

Reply:
I haven't tested, but I guess if refund ever returns null and the component has previous state set, it will not reset back to null. So, this ensures it never happens, and is a good precaution for resettings the state.

Reply:
Done